scRNA-seq-advanced/05-aucell.Rmd

```diff
@@ -0,0 +1,403 @@
+processed_dir <- fs::path(ewing_data_dir, "processed")
+
+# Directory for holding pathway analysis results
+analysis_dir <- fs::path("analysis", "ewing-sarcoma", "pathway-analysis")
+# Create if it doesn't exist yet
+fs::dir_create(analysis_dir)
+```
+
+#### Files
+
+The input will be a `SingleCellExperiment` for an individual Ewing sarcoma library.
+
+```{r setup_input_files}
+sce_file <- fs::path(processed_dir, 
+                     "SCPCS000490", 
+                     "SCPCL000822_processed.rds")
+```
+
+We will save the AUCell results as a table in the analysis directory.
+
+```{r setup_output_files, live = TRUE}
+output_file <- fs::path(analysis_dir,
+                        "ewing_sarcoma_aucell_results.tsv")
+```
+
+
+### Functions
+
+The `source()` function allows us to load in custom functions we saved in an `.R` file.
+
+```{r source_functions}
+source(fs::path("util", "aucell_functions.R"))
+```
+
+This loads one custom function, called `plot_recovery_curve()`, into our environment.
+This function is adapted from [the AUCell vignette](https://github.com/aertslab/AUCell/blob/91753b327a39dc7a4bbed46408ec2271c485f2f0/vignettes/AUCell.Rmd#L295-L316).
+
+## Set up gene sets
+
+We are going to use two gene sets pertaining to Ewing sarcoma.
+
+* [`ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION`](https://www.gsea-msigdb.org/gsea/msigdb/geneset_page.jsp?geneSetName=ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION), which are genes that were highly expressed in a rhabdomyosarcoma cell line engineered to express the EWSR1-FLI1 fusion.
+* [`RIGGI_EWING_SARCOMA_PROGENITOR_UP`](https://www.gsea-msigdb.org/gsea/msigdb/cards/RIGGI_EWING_SARCOMA_PROGENITOR_UP), which are genes that were highly expressed in mesenchymal stem cells engineered to express the EWS-FLI1 fusion protein.
+
+We would expect both of these gene sets to have high expression in tumor cells.
+
+```{r genesets}
+ewing_gene_set_names <- c("ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION",
+                          "RIGGI_EWING_SARCOMA_PROGENITOR_UP")
+```
+
+These gene sets come from the C2 gene set collection from MSigDB.
+Let's retrieve them using `msigdbr()`.
+
+```{r extract_genesets, live = TRUE}
+ewing_gene_sets_df <- msigdbr(species = "Homo sapiens",
+                              category = "C2",
+                              subcategory = "CGP") |>
+  dplyr::filter(gs_name %in% ewing_gene_set_names)
+```
+
+`AUCell` uses gene sets in a particular format that comes from the `GSEABase` package.
+We need to create a `GeneSetCollection`.
+
+```{r gene_set_collection}
+ewing_gene_set_collection <- ewing_gene_set_names |>
+  purrr::map(
+    # For each gene set
+    \(gene_set_name) ewing_gene_sets_df |>
+      # Subset to the rows in that gene set
+      dplyr::filter(gs_name == gene_set_name) |>
+      # Grab the Ensembl gene identifiers
+      dplyr::pull(ensembl_gene) |>
+      # Create a GeneSet object
+      GeneSet(setName = gene_set_name,
+              geneIdType = ENSEMBLIdentifier())
```

Comment:
```suggestion
    \(gene_set_name) {
    ewing_gene_sets_df |>
      # Subset to the rows in that gene set
      dplyr::filter(gs_name == gene_set_name) |>
      # Grab the Ensembl gene identifiers
      dplyr::pull(ensembl_gene) |>
      # Create a GeneSet object
      GeneSet(setName = gene_set_name,
              geneIdType = ENSEMBLIdentifier())
              }
```
Minor formatting but I think these can be easier to read if you start a new line and use the curly brackets to enclose anything that's happen within `purrr::map`.

Reply:
I don't know if the GitHub suggestion will be exactly right, so you may need to adjust it.